src/Uno.UWP/Storage/Pickers/FileOpenPicker.iOS.cs

```diff
@@ -65,13 +65,39 @@ private UIViewController GetViewController(bool multiple, int limit, TaskComplet
 			{
 				case PickerLocationId.PicturesLibrary when multiple is false || iOS14AndAbove is false:
 				case PickerLocationId.VideosLibrary when multiple is false || iOS14AndAbove is false:
-					return new UIImagePickerController()
+					var controller = new UIImagePickerController()
```

Comment:
UIImagePickerController does not support filtering the content based on the extensions; instead, we have to do a couple of steps.

- First, validate the available MediaTypes based on the source (Image and Movie)
- Then, using the list of extensions provided by the user, match this to either image or video
- If either (or both) video and image extensions were provided, then we will include them in the filter

Note: if the filter contains "*", we will include all the possible options (this is as it was before)